opencog/cython/PythonEval.cc

```diff
@@ -1055,6 +1105,8 @@ void PythonEval::apply_as(const std::string& moduleFunction,
 
     // Get the expected argument count.
     int expectedArgumentCount = this->argument_count(pyUserFunc);
+    // If there is an object, its method has 'self' argument, which is passed automatically
+    if (pyObject) expectedArgumentCount--;
```

Comment:
Could fail on static methods, i guess.. Python3 has PyInstanceMethod_Check to check if it is instance method. On the other hand there is PyMethod_GET_SELF for python3 and  python2 as well which returns nullptr if method doesn't has self.

Reply:
@noskill . Good point, but both these functions don't work. `PyInstanceMethod_Check` always returns null, while `PyMethod_GET_SELF` always returns something.

Reply:
Sorry for misleading you. I found that **PyMethod_Check** works, PyInstanceMethod_Check checks for something else and PyMethod_GET_SELF is made to return something anyway.

Reply:
I suggest to change this line to:
```c++
    if (pyObject and PyMethod_Check(pyUserFunc)){
            expectedArgumentCount--;
    }  
```